src/sys/signal.rs

```diff
@@ -92,12 +92,27 @@ impl SigSet {
         Errno::result(res).map(drop)
     }
 
+    pub fn clear(&mut self) -> Result<()> {
+        let res = unsafe { libc::sigemptyset(&mut self.sigset as *mut libc::sigset_t) };
+
+        Errno::result(res).map(drop)
+    }
+
     pub fn remove(&mut self, signum: SigNum) -> Result<()> {
         let res = unsafe { libc::sigdelset(&mut self.sigset as *mut libc::sigset_t, signum) };
 
         Errno::result(res).map(drop)
     }
 
+    pub fn union(&mut self, other: &SigSet) -> Result<()> {
+        for i in 1..NSIG {
+            if try!(other.contains(i)) {
+                try!(self.add(i));
+            }
```

Comment:
I'm surprised this is a loop! I don't know if we want what looks like 1-2 bitwise operations to actually end up being ~50-100. My feeling is that at this level, small code should have small cost.